pkg/api/response/response.go

```diff
@@ -187,6 +188,26 @@ func JSONStreaming(status int, body interface{}) StreamingResponse {
 	}
 }
 
+// JSONDownload creates a JSON response indicating that it should be downloaded.
+func JSONDownload(status int, body interface{}, filename string) *NormalResponse {
+	return JSON(status, body).SetHeader("Content-Disposition", fmt.Sprintf(`attachment;filename="%s"`, filename))
+}
+
+// YAML creates a YAML response.
+func YAML(status int, body interface{}) *NormalResponse {
+	b, err := yaml.Marshal(body)
+	if err != nil {
+		return Error(http.StatusInternalServerError, "body yaml marshal", err)
+	}
+	// As of now, application/yaml is downloaded by default in chrome regardless of Content-Disposition, so we use text/yaml instead.
+	return Respond(status, b).SetHeader("Content-Type", "text/yaml")
```

Comment:
Should this be `string(b)` so the response isn't an array of bytes?

```suggestion
	return Respond(status, string(b)).SetHeader("Content-Type", "text/yaml")
```

Reply:
`Respond` handles multiple types (`[]byte`, `string`, and `struct`) so it will correctly handle this here. It might have been cleaner in to modify `Respond` to handle `yaml` marshalling internally as an option but I wanted to avoid modifying existing methods in this file as much as possible since it's used in so many unrelated places.

Reply:
@JohnnyQQQQ 

> LGTM, there is one change that I'm missing and it's about the documentation. We should update the documentation [here](https://github.com/grafana/grafana/blob/db51e963de5cd362cd52d24462b33c237c8f8514/docs/sources/alerting/set-up/provision-alerting-resources/file-provisioning/index.md#L32) to use the new endpoint.

Updated api markdown and docs to use export endpoints here: ac1a3f1be0176d35569c4be33e93b255789221c4 and df5b5a242aa31fbb2481485a4b3feed1d4a55335